File: Components/DevExpressVCL/Demos/ExpressQuantumGrid/Delphi/IssueList/IssueListForm.pas
unit IssueListForm;

interface

uses
  Windows, Messages, Forms, SysUtils, Classes, DBCtrls, Controls, ExtCtrls,
  StdCtrls, IssueListGrid, cxControls, cxNavigator, cxGraphics, cxLookAndFeels,
  cxLookAndFeelPainters, cxContainer, cxEdit, cxLabel;

type

  TFormClass = class of TfrmBasic;

  TfrmBasic = class(TForm)
    lbCaption: TcxLabel;
    plTop: TPanel;
    cxNavigator: TcxNavigator;
  private
    FGridForm: TIssueListGridForm;
    function GetCaption: string;
    procedure SetCaption(ACaption: string);
  public
    constructor Create(AOwner: TComponent); override;
    property Caption: string read GetCaption write SetCaption;
    property GridForm: TIssueListGridForm read FGridForm;
  end;

implementation

{$R *.dfm}

uses
  IssueListData, IssueListMain;

{ frmBasic }

constructor TfrmBasic.Create(AOwner: TComponent);
begin
  inherited Create(AOwner);
  Visible := False;
  Align := alClient;
  FGridForm := IssueListMainForm.GridForm;
end;

function TfrmBasic.GetCaption: string;
begin
  Result := lbCaption.Caption;
end;

procedure TfrmBasic.SetCaption(ACaption: string);
begin
  lbCaption.Caption := '  ' + ACaption;
end;

end.

File: Components/DevExpressVCL/Demos/ExpressQuantumGrid/CBuilder/IssueList/IssueListForm.cpp
//---------------------------------------------------------------------------

#include <vcl.h>
#pragma hdrstop

#include "IssueListForm.h"
#include "IssueListMain.h"
#include "IssueListGrid.h"

//---------------------------------------------------------------------------
#pragma package(smart_init)
#pragma link "cxControls"
#pragma link "cxNavigator"
#pragma link "cxContainer"
#pragma link "cxEdit"
#pragma link "cxGraphics"
#pragma link "cxLabel"
#pragma link "cxLookAndFeelPainters"
#pragma link "cxLookAndFeels"
#pragma resource "*.dfm"
TfrmBasic *frmBasic;
//---------------------------------------------------------------------------
__fastcall TfrmBasic::TfrmBasic(TComponent* Owner)
  : TForm(Owner)
{
  Visible = false;
  Align = alClient;
  FGridForm = IssueListMainForm->GridForm;
}
//---------------------------------------------------------------------------

String TfrmBasic::GetCaption()
{
  return lbCaption->Caption;
}
//---------------------------------------------------------------------------
void TfrmBasic::SetCaption(String ACaption)
{
  lbCaption->Caption = "  " + ACaption;
}
//---------------------------------------------------------------------------

